control-plane/grpc/src/operations/volume/traits.rs

```diff
@@ -1630,3 +1631,40 @@ impl TryFrom<volume::AffinityGroupSpec> for AffinityGroupSpec {
         Ok(AffinityGroupSpec::new(value.id, volumes))
     }
 }
+
+// Create volume creation event message from volume data.
+impl Event for Volume {
+    fn event(&self) -> EventMessage {
+        let node_id = match self.spec().target() {
+            Some(target) => target.node().to_string(),
+            None => "".to_string(),
+        };
+        let event_meta = EventMeta::new();
+        let event_source = EventSource::new();
+        EventMessage {
+            category: EventCategory::Volume as i32,
+            action: EventAction::Create as i32,
+            target: self.uuid().to_string(),
+            metadata: Some(EventMeta {
+                source: Some(EventSource {
+                    node: node_id,
+                    ..event_source
+                }),
+                ..event_meta
+            }),
+        }
+    }
+}
+
+// Create volume delete event message.
+impl Event for ValidatedDestroyVolumeRequest {
```

Comment:
Why are the types implementing 'Event' in case of vol creation and vol deletion named so differently (Volume vs ValidateDestroyVolumeRequest)

Reply:
I could get the volume creation data from 'volume' and deletion data from 'ValidateDestroyVolumeRequest'. So I implemented event trait on these types to get the EventMessage.